plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/TransformationMatrix.cpp

```diff
@@ -43,6 +43,21 @@ MPxTransformationMatrix* TransformationMatrix::creator()
   return new TransformationMatrix;
 }
 
+//----------------------------------------------------------------------------------------------------------------------
+bool hasEmptyDefaultValue(const UsdGeomXformOp& op, UsdTimeCode time)
```

Comment:
I'm not sure about the actual c++ rule but I always prefer to be explicit and wrap these local functions in an unnamed namespace{}.

Was also thinking that we could just make _hasEmptyDefaultValue_ do everything you have below.  It's only used in that exact same chunk of code in the 5 or 6 places.  Why not just have it also check "timeCode.IsDefault() && op.GetNumTimeSamples()"